core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -118,8 +116,6 @@ class LogManager(logDirs: Seq[File],
     logDirsSet
   }
 
-  loadLogs()
```

Comment:
Looks like we need to look if it's still safe for the things after this to be run in the constructor. Perhaps the cleaner should be started after the logs are loaded?

Reply:
Good point.  Yes, it seems safer to delay starting anything until we actually have recovered.